viewing_party/party.py

```diff
@@ -1,23 +1,201 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    new_movie = {}
+    
+    if not title or not genre or not rating:
+        return None
+    
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    return new_movie
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"] += [movie]

```

Comment:
This is list concatting here and this actually creates a new list each time it is done, so this would actually be have a time complexity of `O(n)`, whereas using the `append` method would give us a time complexity of `O(1)`. This implementation would also take your space complexity from `O(1)` to `O(n)`. This was a clever and creative implementation!